### PR TITLE
fix(polyfills-loader): fix #1800 Add ie11 compatibility for polyfillsLoader

### LINKED ISSUE #1800: Polyfill loader injects modern script
Looking at [https://modern-web.dev/docs/building/rollup-plugin-polyfills-loader/](https://modern-web.dev/docs/building/rollup-plugin-polyfills-loader/)

the output of the polyfill loader is "modern" javascript and can therefore not be executed on IE11

Example of rollup build:

```
// Import rollup plugins
import html from '@web/rollup-plugin-html';
import polyfillsLoader from '@web/rollup-plugin-polyfills-loader';
import {copy} from '@web/rollup-plugin-copy';
import resolve from '@rollup/plugin-node-resolve';
import {getBabelOutputPlugin} from '@rollup/plugin-babel';
import {terser} from 'rollup-plugin-terser';
import minifyHTML from 'rollup-plugin-minify-html-literals';
import summary from 'rollup-plugin-summary';

// Configure an instance of @web/rollup-plugin-html
const htmlPlugin = html({
  rootDir: './',
  flattenOutput: false,
});

export default {
  // Entry point for application build; can specify a glob to build multiple
  // HTML files for non-SPA app
  input: 'index.html',
  plugins: [
    htmlPlugin,
    // Resolve bare module specifiers to relative paths
    resolve(),
    // Minify HTML template literals
    minifyHTML(),
    // Minify JS
    terser({
      module: true,
      warnings: true,
    }),
    // Inject polyfills into HTML (core-js, regnerator-runtime, webcoponents,
    // lit/polyfill-support) and dynamically loads modern vs. legacy builds
    polyfillsLoader({
      modernOutput: {
        name: 'modern',
      },
      // Feature detection for loading legacy bundles
      legacyOutput: {
        name: 'legacy',
        test: "!('noModule' in HTMLScriptElement.prototype)",
        type: 'systemjs',
      },
      // List of polyfills to inject (each has individual feature detection)
      polyfills: {
        webcomponents: false,
        coreJs: true,
        fetch: true,
        // Custom configuration for loading Lit's polyfill-support module,
        // required for interfacing with the webcomponents polyfills
        custom: [
          {
            name: 'webcomponent-loader',
            path: 'node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js',
            module: false,
          },
          {
            name: 'lit-polyfill-support',
            path: 'node_modules/lit/polyfill-support.js',
            test: "!('attachShadow' in Element.prototype)",
            module: false,
          },
        ],
      },
    }),
    // Print bundle summary
    summary(),
    // Optional: copy any static assets to build directory
    copy({
      patterns: ['data/**/*', 'images/**/*'],
    }),
  ],
  // Specifies two JS output configurations, modern and legacy, which the HTML plugin will
  // automatically choose between; the legacy build is compiled to ES5
  // and SystemJS modules
  output: [
    {
      // Modern JS bundles (no JS compilation, ES module output)
      format: 'esm',
      chunkFileNames: '[name]-[hash].js',
      entryFileNames: '[name]-[hash].js',
      dir: 'build',
      plugins: [htmlPlugin.api.addOutput('modern')],
    },
    {
      // Legacy JS bundles (ES5 compilation and SystemJS module output)
      format: 'esm',
      chunkFileNames: 'legacy-[name]-[hash].js',
      entryFileNames: 'legacy-[name]-[hash].js',
      dir: 'build',
      plugins: [
        htmlPlugin.api.addOutput('legacy'),
        // Uses babel to compile JS to ES5 and modules to SystemJS
        getBabelOutputPlugin({
          compact: true,
          presets: [
            [
              '@babel/preset-env',
              {
                targets: {
                  ie: '11',
                },
                modules: 'systemjs',
              },
            ],
          ],
        }),
      ],
    },
  ],
  preserveEntrySignatures: false,
};
```

This produces the following output:
```
  function polyfillsLoader() {
    function loadScript(src, type, attributes = []) {
      return new Promise(function (resolve) {
        var script = document.createElement('script');

        function onLoaded() {
          if (script.parentElement) {
            script.parentElement.removeChild(script);
          }

          resolve();
        }

        script.src = src;
        script.onload = onLoaded;
        attributes.forEach(att => {
          script.setAttribute(att.name, att.value);
        });

```

The issue is the arrow function and of course the attributes = [], which is not allowed in IE 11

### PR DESCRIPTION
**Pull request by twigy11**
… script not compatible with ie11

## What I did

1. Change the loadScriptFunction script to be available to run in ie11.

### CODE REVIEW

File: packages/polyfills-loader/src/createPolyfillsLoader.ts [hunk 1]
```diff
@@ -29,9 +29,11 @@ const loadScriptFunction = `
       }
       script.src = src;
       script.onload = onLoaded;
-      attributes.forEach(att => {
-        script.setAttribute(att.name, att.value);
-      });
+      if(attributes){
```
Comment (web-padawan) [inline packages/polyfills-loader/src/createPolyfillsLoader.ts:32]:
```suggestion
      if (Array.isArray(attributes)) {
```

File: packages/polyfills-loader/src/createPolyfillsLoader.ts [hunk 2]
```diff
@@ -29,9 +29,11 @@ const loadScriptFunction = `
...
+        attributes.forEach(att => {
```
Comment (web-padawan) [inline packages/polyfills-loader/src/createPolyfillsLoader.ts:33]: This needs to be changed to ES5 as well:

```suggestion
        attributes.forEach(function (att) {
```

**Review by web-padawan — CHANGES_REQUESTED** [review]
Please update snapshots for this change and include them to rhis PR. You can do it like this:

```sh
node scripts/workspaces-scripts-bin.mjs test:update-snapshots
```

File: packages/polyfills-loader/src/createPolyfillsLoader.ts (hunk 2 above)
Comment (twigy11) [reply packages/polyfills-loader/src/createPolyfillsLoader.ts:33]: ups.. I've forgotten to copy this change. Totally agree about that

**Comment by twigy11** [comment]
hi @web-padawan , I'm trying to update the snapshot as you told me, but I always get the same error.

Failed to execute command test:update-snapshots for 3 packages. But we don't know which ones, because concurrently doesn't say

- First error

**[rollup-plugin-polyfills-loader] TSError: ⨯ Unable to compile TypeScript:
[rollup-plugin-polyfills-loader] src/createPolyfillsLoaderConfig.ts(1,10): error TS2305: Module 
'"../../../node_modules/@web/rollup-plugin-html"' has no exported member 'EntrypointBundle'.
[rollup-plugin-polyfills-loader] src/createPolyfillsLoaderConfig.ts(2,10): error TS2305: Module '"../../../node_modules/@web/polyfills-loader"' has no exported member 'PolyfillsLoaderConfig'.
[rollup-plugin-polyfills-loader] src/createPolyfillsLoaderConfig.ts(2,33): error TS2305: Module '"../../../node_modules/@web/polyfills-loader"' has no exported member 'FileType'.
[rollup-plugin-polyfills-loader] src/createPolyfillsLoaderConfig.ts(2,43): error TS2305: Module '"../../../node_modules/@web/polyfills-loader"' has no exported member 'fileTypes'.
[rollup-plugin-polyfills-loader] src/createPolyfillsLoaderConfig.ts(30,40): error TS7006: Parameter 'e' implicitly has an 'any' type.**

- second error. This error is the same when I'm trying to execute the test in the specific change package

**c/utils.ts(2,10): error TS2305: Module '"../../../node_modules/@web/parse5-utils"' has no exported member 'getAttribute'.**

I've attached the complete trace
[test.txt](https://github.com/modernweb-dev/web/files/8365831/test.txt)

I don't know if you can help me with this issue.

Thank you in advance

**Comment by twigy11** [comment]
I understood whats going on. The project doesn't take the package folder when trying to resolve the bare module import. In my local branch, I've changed it  for relative path and I've could run the test successfully.

So, Are any documentation for configure my enviroment. I've followed the contribution documentation but I didn't get it. Any Idea?

Thank you in advance, and sorry for all questions, it's my first contribution time and I'm a little bit lost :)

**Comment by LarsDenBakker** [comment]
Thanks for your PR, I included your fix in https://github.com/modernweb-dev/web/pull/1926 and updated the snapshots and fixed linting errors.

**Pull request closed without merging**